vendor/assets/javascripts/select2.min.js

```diff
@@ -833,8 +833,8 @@ the specific language governing permissions and limitations under the Apache Lic
                     process=function(element, collection) {
                         var group;
                         if (element.is("option")) {
-                            if (query.matcher(term, (element.attr('alias')) ? element.attr('alias') : element.text(), element)) {
-                                collection.push({id:element.attr("value"), text:element.text(), element: element.get(), css: element.attr("class"), alias: element.attr('alias')});
+                            if (query.matcher(term, (element.attr('alias') !== 'undefined') ? element.attr('alias') : element.text(), element)) {
```

Comment:
I'm not sure we can modify JQuery library (licensing).

We'll have to use another way :)